src/plugins/dist_matrixops/all_gather.cpp

```diff
@@ -0,0 +1,383 @@
+// Copyright (c) 2019 Shahrzad Shirzad
+// Copyright (c) 2020 Hartmut Kaiser
+// Copyright (c) 2020 Nanmiao Wu
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <phylanx/config.hpp>
+#include <phylanx/execution_tree/annotation.hpp>
+#include <phylanx/execution_tree/locality_annotation.hpp>
+#include <phylanx/execution_tree/localities_annotation.hpp>
+#include <phylanx/execution_tree/meta_annotation.hpp>
+#include <phylanx/execution_tree/primitives/annotate_primitive.hpp>
+#include <phylanx/execution_tree/primitives/node_data_helpers.hpp>
+#include <phylanx/execution_tree/tiling_annotations.hpp>
+#include <phylanx/ir/node_data.hpp>
+#include <phylanx/plugins/dist_matrixops/all_gather.hpp>
+#include <phylanx/plugins/dist_matrixops/tile_calculation_helper.hpp>
+#include <phylanx/plugins/matrixops/concatenate.hpp>
+#include <phylanx/util/distributed_matrix.hpp>
+#include <phylanx/util/distributed_vector.hpp>
+#include <phylanx/util/generate_error_message.hpp>
+#include <phylanx/util/index_calculation_helper.hpp>
+
```

Comment:
I guess some of these includes are not needed; I didn't find utilization for `tile_calculation_helper` or `index_calculation_helper`

Reply:
Yes. Will remove them.